src/pages/studyView/StudyViewPage.tsx

```diff
@@ -354,6 +419,42 @@ export default class StudyViewPage extends React.Component<
         );
     }
 
+    readonly openResourceTabs = MakeMobxView({
+        await: () => [
+            this.store.resourceDefinitions,
+            this.store.resourceIdToResourceData,
+        ],
+        render: () => {
+            const openDefinitions = this.store.resourceDefinitions.result!.filter(
+                d => this.store.isResourceTabOpen(d.resourceId)
+            );
+            const sorted = _.sortBy(openDefinitions, d => d.priority);
+            const resourceDataById = this.store.resourceIdToResourceData
+                .result!;
+
+            const tabs: JSX.Element[] = [];
```

Comment:
i've been telling everyone to use reduce for these kind of filteredMap operations.  or do a filter and then a map.   i just think its more readable.   what do you think?  i know there are lots of these in this PR but i just decided to put comment down here.